# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/2d/2D.h
# Commit: de6ddf5eb2a7
# Full Hash: de6ddf5eb2a7bd3d8a2f58636760e0aa16c584b9
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 773e7b31ac6e -r de6ddf5eb2a7 gfx/2d/2D.h
--- a/gfx/2d/2D.h	Tue Dec 05 09:37:03 2023 +0000
+++ b/gfx/2d/2D.h	Tue Dec 05 09:37:03 2023 +0000
@@ -700,6 +700,16 @@
     return surface.forget();
   }
 
+  /**
+   * Attempts to read the contents of this surface into a destination data
+   * surface. This may simply return false to signal that this operation is
+   * not supported or is not efficient, in which case the caller should try
+   * to convert this surface to a data surface to read from it instead.
+   */
+  virtual bool ReadInto(DataSourceSurface* aSurface, const IntRect& aRect) {
+    return false;
+  }
+
   /** Tries to get this SourceSurface's native surface.  This will fail if aType
    * is not the type of this SourceSurface's native surface.
    */